chordv-0.1/Doxygen/man/man3/FormConfig.3
.TH "FormConfig" 3 "Sun Apr 15 2018" "Version 0.1" "Chord V" \" -*- nroff -*-
.ad l
.nh
.SH NAME
FormConfig \- The \fBFormConfig\fP class manage the display of configuration of attributes : (Color, Fonts, \&.\&.\&.) for the definition of each booklet\&. It is used in Preference and also on each project\&.  

.SH SYNOPSIS
.br
.PP
.PP
\fC#include <formconfig\&.h>\fP
.PP
Inherits QWidget\&.
.PP
Inherited by \fBChordConfig\fP, \fBLyricsConfig\fP, \fBMemoryConfig\fP, and \fBTextConfig\fP\&.
.SS "Public Types"

.in +1c
.ti -1c
.RI "enum \fBClasses\fP { \fBChord\fP, \fBLyrics\fP, \fBMemory\fP, \fBText\fP }
.RI "Classes is different for each style of booklet\&. ""
.br
.ti -1c
.RI "enum \fBChordInclusion\fP { \fBDiagramInText\fP, \fBChordAtBeginingAndChordNameInText\fP, \fBNoChordAtBeginingAndChordNameInText\fP }
.RI "ChordInclusion has different capabilities for Text and Lyrics\&. ""
.br
.in -1c
.SS "Signals"

.in +1c
.ti -1c
.RI "void \fBsendLog\fP (QString message)"
.br
.in -1c
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBFormConfig\fP (QWidget *parent=0)"
.br
.RI "\fBFormConfig\fP constructor complient with QtDesigner allowing promotion of QWidget\&. "
.ti -1c
.RI "\fB~FormConfig\fP ()"
.br
.ti -1c
.RI "void \fBsetValue\fP (QString var, QVariant value)"
.br
.RI "setValue allow to set a Value for an object it is based on the name of the object for example comboBoxName => Name is extracted and QSettings is used to Name value The object types analyzed are \fBSpinBoxUnit\fP,\fBFontButton\fP,Font, FontColor,FontBackgroundColor,Cover,comboBox, spinBox,doubleSpinBox,Image,MediaBox,colorButton "
.ti -1c
.RI "void \fBSave\fP (QString filename, \fBClasses\fP section)"
.br
.RI "Save all the \fBSpinBoxUnit\fP,\fBFontButton\fP,Font, FontColor,FontBackgroundColor,Cover,comboBox, spinBox,doubleSpinBox,Image,MediaBox,colorButton extracted name presented in setValue in a filename given in argument\&. The value is the classname : \fBChord\fP, Memory, Lyrics or Text\&. "
.ti -1c
.RI "void \fBInitDefault\fP (\fBClasses\fP classe)"
.br
.RI "InitDefault init all the \fBSpinBoxUnit\fP,\fBFontButton\fP,Font, FontColor,FontBackgroundColor,Cover,comboBox, spinBox,doubleSpinBox,Image,MediaBox,colorButton extracted name with the values given but preference QDialog\&. "
.ti -1c
.RI "virtual Ui::FormConfig * \fBgetUi\fP ()"
.br
.RI "getUi this virtual function return the ui used on each derived class "
.ti -1c
.RI "void \fBSetConfigFromFile\fP (QString FileName)"
.br
.RI "SetConfigFromFile\&. "
.ti -1c
.RI "void \fBSetConfigFromInit\fP ()"
.br
.ti -1c
.RI "QString \fBclasse2String\fP (\fBClasses\fP name)"
.br
.ti -1c
.RI "void \fBRetranslate\fP ()"
.br
.ti -1c
.RI "void \fBdisableGroup\fP (QString group)"
.br
.ti -1c
.RI "void \fBsetProjectPath\fP (QString path)"
.br
.in -1c
.SS "Protected Member Functions"

.in +1c
.ti -1c
.RI "void \fBdisableWidgets\fP (QRegExp value)"
.br
.ti -1c
.RI "void \fBInitComboBoxChordInclusion\fP (QStringList list)"
.br
.in -1c
.SH "Detailed Description"
.PP 
The \fBFormConfig\fP class manage the display of configuration of attributes : (Color, Fonts, \&.\&.\&.) for the definition of each booklet\&. It is used in Preference and also on each project\&. 
.SH "Member Enumeration Documentation"
.PP 
.SS "enum \fBFormConfig::ChordInclusion\fP"

.PP
ChordInclusion has different capabilities for Text and Lyrics\&. 
.PP
\fBEnumerator\fP
.in +1c
.TP
\fB\fIDiagramInText \fP\fP
Diagram are included in Text\&. 
.TP
\fB\fIChordAtBeginingAndChordNameInText \fP\fP
Diagram are include at begining and in Text\&. 
.TP
\fB\fINoChordAtBeginingAndChordNameInText \fP\fP
No diagram at beginin but in text\&. 
.SS "enum \fBFormConfig::Classes\fP"

.PP
Classes is different for each style of booklet\&. 
.PP
\fBEnumerator\fP
.in +1c
.TP
\fB\fIChord \fP\fP
\fBFormConfig\fP is \fBChord\fP oriented\&. 
.TP
\fB\fILyrics \fP\fP
\fBFormConfig\fP is Lyrics oriented\&. 
.TP
\fB\fIMemory \fP\fP
\fBFormConfig\fP is Memory oriented\&. 
.TP
\fB\fIText \fP\fP
\fBFormConfig\fP is Text only oriented\&. 
.SH "Constructor & Destructor Documentation"
.PP 
.SS "FormConfig::FormConfig (QWidget * parent = \fC0\fP)\fC [explicit]\fP"

.PP
\fBFormConfig\fP constructor complient with QtDesigner allowing promotion of QWidget\&. 
.PP
\fBParameters:\fP
.RS 4
\fIparent\fP 
.RE
.PP

.SS "FormConfig::~FormConfig ()"

.SH "Member Function Documentation"
.PP 
.SS "QString FormConfig::classe2String (\fBClasses\fP name)"

.SS "void FormConfig::disableGroup (QString group)"

.SS "void FormConfig::disableWidgets (QRegExp value)\fC [protected]\fP"

.SS "virtual Ui::FormConfig* FormConfig::getUi ()\fC [inline]\fP, \fC [virtual]\fP"

.PP
getUi this virtual function return the ui used on each derived class 
.PP
\fBReturns:\fP
.RS 4
the Ui::FormConfig pointer on the child class 
.RE
.PP

.SS "void FormConfig::InitComboBoxChordInclusion (QStringList list)\fC [protected]\fP"

.SS "void FormConfig::InitDefault (\fBClasses\fP classe)"

.PP
InitDefault init all the \fBSpinBoxUnit\fP,\fBFontButton\fP,Font, FontColor,FontBackgroundColor,Cover,comboBox, spinBox,doubleSpinBox,Image,MediaBox,colorButton extracted name with the values given but preference QDialog\&. 
.PP
\fBParameters:\fP
.RS 4
\fIclasse\fP : Classes enum representing the section in \fBSettings\fP save fi 
.RE
.PP

.SS "void FormConfig::Retranslate ()"

.SS "void FormConfig::Save (QString filename, \fBClasses\fP section)"

.PP
Save all the \fBSpinBoxUnit\fP,\fBFontButton\fP,Font, FontColor,FontBackgroundColor,Cover,comboBox, spinBox,doubleSpinBox,Image,MediaBox,colorButton extracted name presented in setValue in a filename given in argument\&. The value is the classname : \fBChord\fP, Memory, Lyrics or Text\&. 
.PP
\fBParameters:\fP
.RS 4
\fIfilename\fP : the QString filename to save the value 
.br
\fIsection\fP : Classes enum representing the section in \fBSettings\fP save file 
.RE
.PP

.SS "void FormConfig::sendLog (QString message)\fC [signal]\fP"

.SS "void FormConfig::SetConfigFromFile (QString FileName)"

.PP
SetConfigFromFile\&. 
.PP
\fBParameters:\fP
.RS 4
\fIFileName\fP 
.RE
.PP

.SS "void FormConfig::SetConfigFromInit ()"

.SS "void FormConfig::setProjectPath (QString path)"

.SS "void FormConfig::setValue (QString var, QVariant value)"

.PP
setValue allow to set a Value for an object it is based on the name of the object for example comboBoxName => Name is extracted and QSettings is used to Name value The object types analyzed are \fBSpinBoxUnit\fP,\fBFontButton\fP,Font, FontColor,FontBackgroundColor,Cover,comboBox, spinBox,doubleSpinBox,Image,MediaBox,colorButton 
.PP
\fBParameters:\fP
.RS 4
\fIvar\fP the name of object for example comboBoxName 
.br
\fIvalue\fP : value given to Name extracted for comboBox 
.RE
.PP


.SH "Author"
.PP 
Generated automatically by Doxygen for Chord V from the source code\&.


src/formconfig.cpp
#include "formconfig.h"
#include "ui_formconfig.h"


#include "colorbutton.h"
#include "examplelabel.h"
#include "settings.h"
#include "language.h"
#include "const.h"

#include <QDebug>
#include <QRegExp>
#include <QLineEdit>
#include <QSettings>

FormConfig::FormConfig(QWidget *parent) :
    QWidget(parent),
    ui(new Ui::FormConfig)
{
    ui->setupUi(this);
    SetConfigFromInit();
    Language::setLanguageComboBox(ui->comboBoxChordLang);
    m_parent=parent;
    connect (ui->checkBoxCover,SIGNAL(stateChanged(int)),this,SLOT(setCover(int)));
    foreach (FontButton *w ,m_parent->findChildren<FontButton*>())
    connect (w,SIGNAL(sendSelectedFont(QFont,QColor,QColor)),this,SLOT(displayFont(QFont,QColor,QColor)));
    connect(ui->comboBoxMediaBox,SIGNAL(currentTextChanged(QString)),this,SLOT(SizeChanged(QString)));
    connect(ui->checkBoxLandscape,SIGNAL(clicked(bool)),this,SLOT(SizeChanged(bool)));
    connect(ui->toolButtonCoverImage,SIGNAL(ImageSelelected(QString)),this,SLOT(displayThumb(QString)));
    connect(ui->spuPageHeight,SIGNAL(valueChanged(double)),this,SLOT(FindSize(double)));
    connect(ui->spuPageWidth,SIGNAL(valueChanged(double)),this,SLOT(FindSize(double)));
    connect(ui->toolButtonSuppressCoverImage,SIGNAL(clicked(bool)),this,SLOT(deleteCoverImage(bool)));
    connect (ui->comboBoxPageNumberStyle,SIGNAL(currentIndexChanged(int)),this,SLOT(CheckArrow(int)));
}

void FormConfig::SizeChanged(bool)
{
    Calculate();
}

void FormConfig::SizeChanged(QString)
{
    Calculate();
}


void FormConfig::CheckArrow(int i )
{
    if ( i == Const::NumberAndArrows )
    {
        ui->comboBoxPageNumber->setCurrentIndex(Const::Center);
        ui->comboBoxPageNumber->setDisabled(true);
    }
    else
    {
        ui->comboBoxPageNumber->setDisabled(false);
    }
}

void FormConfig::Calculate()
{
    double height=ui->comboBoxMediaBox->getHeight();
    double width=ui->comboBoxMediaBox->getWidth();
    QString unit=ui->comboBoxMediaBox->getUnit();
     if ( ui->checkBoxLandscape->isChecked() )
     {
         double l=width;
         width=height;
         height=l;
     }
    ui->spuPageHeight->setValue(height,ui->spuPageHeight->string2unit(unit));
    ui->spuPageWidth->setValue(width,ui->spuPageWidth->string2unit(unit));
}

void FormConfig::FindSize(double)
{
    bool landscape;
    QString val=ui->comboBoxMediaBox->findSize(ui->spuPageWidth->getValue(),ui->spuPageHeight->getValue(),landscape);
    ui->comboBoxMediaBox->disconnect();
    ui->checkBoxLandscape->disconnect();
    ui->comboBoxMediaBox->setCurrentTextByValue(val);
    ui->checkBoxLandscape->setChecked(landscape);
    connect(ui->comboBoxMediaBox,SIGNAL(currentTextChanged(QString)),this,SLOT(SizeChanged(QString)));
    connect(ui->checkBoxLandscape,SIGNAL(clicked(bool)),this,SLOT(SizeChanged(bool)));
}

FormConfig::~FormConfig()
{
    delete ui;
}


void FormConfig::disableWidgets(QRegExp value)
{
    foreach (QLineEdit *w ,m_parent->findChildren<QLineEdit*>(value))
        w->setVisible(false);
    foreach (QLabel *w ,m_parent->findChildren<QLabel*>(value))
        w->setVisible(false);
    foreach (QToolButton *w ,m_parent->findChildren<QToolButton*>(value))
        w->setVisible(false);
    foreach (SpinBoxUnit *w ,m_parent->findChildren<SpinBoxUnit*>(value))
        w->setVisible(false);
    foreach (QCheckBox *w ,m_parent->findChildren<QCheckBox*>(value))
        w->setVisible(false);
    foreach (QComboBox *w ,m_parent->findChildren<QComboBox*>(value))
        w->setVisible(false);

}

void FormConfig::disableGroup(QString group)
{
    foreach (QGroupBox *w ,m_parent->findChildren<QGroupBox*>(group))
        w->setVisible(false);
}

void FormConfig::setProjectPath(QString path)
{
    m_projectpath=path;
    ui->toolButtonCoverImage->setProjectPath(path);
}

void FormConfig::setCover(int val)
{

    ui->labelCoverImage->setEnabled(val);
    ui->toolButtonCoverImage->setEnabled(val);
    ui->comboBoxTitlePosition->setEnabled(val);
    ui->toolButtonSuppressCoverImage->setEnabled(val);
}



void FormConfig::setValue(QString var, QVariant value)
{
    QRegExp spu ("^spu(.*)") ;
    if ( var.contains(spu))
    {
        foreach (SpinBoxUnit *w , m_parent->findChildren<SpinBoxUnit*>(var)) w->setValue(value.toString());

    }
    if ( var.endsWith("Font"))
    {
        QFont f;
        f.fromString(value.toString());
        foreach (FontButton *w ,m_parent->findChildren<FontButton*>(QString("toolButton")+var))  w->setFont(f);
    }
    else if ( var.endsWith("FontColor"))
    {
        foreach (FontButton *w ,m_parent->findChildren<FontButton*>(QString("toolButton")+var))  w->setColor(QColor(value.toString()));
    }
    else if ( var.endsWith("FontBackgroundColor"))
    {
        foreach (FontButton *w ,m_parent->findChildren<FontButton*>(QString("toolButton")+var))  w->setBackgroundColor(QColor(value.toString()));
    }
    else if ( var==QString("Cover") )
    {
         ui->checkBoxCover->setChecked(value.toInt()==1);
    }
  else if ( var.startsWith("comboBox") )
    {
            foreach (QComboBox *w , m_parent->findChildren<QComboBox*>(var)) w->setCurrentText(value.toString());
    }
    else if ( var.startsWith("spinBox"))
    {
           foreach (QSpinBox *w ,m_parent->findChildren<QSpinBox*>(var)) w->setValue(value.toInt());

    }
    else if ( var.startsWith("doubleSpinBox") )
    {
           foreach (QDoubleSpinBox *w ,m_parent->findChildren<QDoubleSpinBox*>(var)) w->setValue(value.toDouble());

    }
    else if ( var.endsWith("Image"))
    {

        foreach (ImageButton *w ,m_parent->findChildren<ImageButton*>(QString("toolButton")+var))
            w->setImage(value.toString());

    }
    else if ( var.endsWith("MediaBox"))
    {
        foreach (PageSize *w ,m_parent->findChildren<PageSize*>(QString("comboBoxMediaBox")))
        {
            w->setCurrentText(value.toString());
        }

    }
   else if ( var.startsWith("colorButton"))
    {
       foreach (ColorButton *w ,m_parent->findChildren<ColorButton*>(QString("colorButton")+var))
            {
                w->setColor(QColor(value.toString()));
            }

    }
   // else emit sendLog(QString ("Notice: (à finir) %1 => %2").arg(var).arg(value.toString()));

}


void FormConfig::InitDefault(Classes c)
{

    QString classe=classe2String(c);
    Settings *s;
    if( m_configFileName.isEmpty()) s = new Settings() ;
    else s= new Settings(m_configFileName);
    ui->checkBoxCover->setChecked(s->value(QString("%1/Cover").arg(classe),true).toBool());
    ui->checkBoxFullScreenMode->setChecked(s->value(QString("%1/FullScreenMode").arg(classe),false).toBool());
    ui->checkBoxTitleInUppercase->setChecked(s->value(QString("%1/TitleInUppercase").arg(classe),false).toBool());

    ui->comboBoxChordInText->setCurrentIndex(s->value(QString("%1/ChordInText").arg(classe),0).toInt());
    ui->comboBoxChordLang->setCurrentIndex(s->value(QString("%1/ChordLang").arg(classe),"0").toInt());
    ui->comboBoxTocColumnNumber->setCurrentIndex(s->value(QString("%1/TocColumnNumber").arg(classe),0).toInt());
    ui->comboBoxTocSpaceCharacter->setCurrentIndex(s->value(QString("%1/TocSpaceCharacter").arg(classe),0).toInt());
    ui->comboBoxTocPosition->setCurrentIndex(s->value(QString("%1/TocPosition").arg(classe),1).toInt());
    ui->comboBoxTitlePosition->setCurrentIndex(s->value(QString("%1/TitlePosition").arg(classe),"0").toInt());
    ui->comboBoxDuplex->setCurrentIndex(s->value(QString("%1/Duplex").arg(classe),0).toInt());
    ui->comboBoxPageNumber->setCurrentIndex(s->value(QString("%1/PageNumber").arg(classe),1).toInt());
    ui->comboBoxTocVerticalSpacing->setCurrentIndex(s->value(QString("%1/TocVerticalSpacing").arg(classe),1).toInt());
    ui->comboBoxPageNumberStyle->setCurrentIndex(s->value(QString("%1/PageNumberStyle").arg(classe),1).toInt());

    ui->lineEditOutFile->setText(s->value(QString("%1/OutFile").arg(classe),classe).toString());
    ui->spuChordHorizontalSize->setValue(s->value(QString("%1/ChordHorizontalSize").arg(classe),"2mm").toString());
    ui->spuHorizontalMargin->setValue(s->value(QString("%1/HorizontalMargin").arg(classe),"2mm").toString());
    ui->spuVerticalMargin->setValue(s->value(QString("%1/VerticalMargin").arg(classe),"5mm").toString());

    ui->spuPageHeight->setValue(s->value(QString("%1/PageHeight").arg(classe),"297mm").toString());
    ui->spuPageWidth->setValue(s->value(QString("%1/PageWidth").arg(classe),"210mm").toString());
    QFont f;
    f.fromString(s->value(QString("%1/ChordFont").arg(classe),QFont().toString()).toString());
    ui->toolButtonChordFont->setFont(f);
    f.fromString(s->value(QString("%1/CoverFont").arg(classe),QFont().toString()).toString());
    ui->toolButtonCoverFont->setFont(f);
    f.fromString(s->value(QString("%1/NormalFont").arg(classe),QFont().toString()).toString());
    ui->toolButtonNormalFont->setFont(f);
    f.fromString(s->value(QString("%1/SubtitleFont").arg(classe),QFont().toString()).toString());
    ui->toolButtonSubtitleFont->setFont(f);
    f.fromString(s->value(QString("%1/TitleFont").arg(classe),QFont().toString()).toString());
    ui->toolButtonTitleFont->setFont(f);
    f.fromString(s->value(QString("%1/TocFont").arg(classe),QFont().toString()).toString());
    ui->toolButtonTocFont->setFont(f);
    f.fromString(s->value(QString("%1/PageNumberFont").arg(classe),QFont().toString()).toString());
    ui->toolButtonPageNumberFont->setFont(f);
    f.fromString(s->value(QString("%1/CommentFont").arg(classe),QFont().toString()).toString());
    ui->toolButtonCommentFont->setFont(f);
    f.fromString(s->value(QString("%1/CommentItalicFont").arg(classe),QFont().toString()).toString());
    ui->toolButtonCommentItalicFont->setFont(f);
    f.fromString(s->value(QString("%1/CommentBoxFont").arg(classe),QFont().toString()).toString());
    ui->toolButtonCommentBoxFont->setFont(f);

    ui->toolButtonChordFont->setColor(QColor(s->value(QString("%1/ChordFontColor").arg(classe),"black").toString()));
    ui->toolButtonCoverFont->setColor(QColor(s->value(QString("%1/CoverFontColor").arg(classe),"black").toString()));
    ui->toolButtonNormalFont->setColor(QColor(s->value(QString("%1/NormalFontColor").arg(classe),"black").toString()));
    ui->toolButtonSubtitleFont->setColor(QColor(s->value(QString("%1/SubtitleFontColor").arg(classe),"black").toString()));
    ui->toolButtonTitleFont->setColor(QColor(s->value(QString("%1/TitleFontColor").arg(classe),"black").toString()));
    ui->toolButtonPageNumberFont->setColor(QColor(s->value(QString("%1/PageNumberFontColor").arg(classe),"black").toString()));
    ui->toolButtonTocFont->setColor(QColor(s->value(QString("%1/TocFontColor").arg(classe),"black").toString()));
    ui->toolButtonCommentFont->setColor(QColor(s->value(QString("%1/CommentFontColor").arg(classe),"black").toString()));
    ui->toolButtonCommentItalicFont->setColor(QColor(s->value(QString("%1/CommentItalicFontColor").arg(classe),"black").toString()));
    ui->toolButtonCommentBoxFont->setColor(QColor(s->value(QString("%1/CommentBoxFontColor").arg(classe),"black").toString()));

    ui->toolButtonChordFont->setBackgroundColor(QColor(s->value(QString("%1/ChordFontBackgroundColor").arg(classe),"white").toString()));
    ui->toolButtonCoverFont->setBackgroundColor(QColor(s->value(QString("%1/CoverFontBackgroundColor").arg(classe),"white").toString()));
    ui->toolButtonNormalFont->setBackgroundColor(QColor(s->value(QString("%1/NormalFontBackgroundColor").arg(classe),"white").toString()));
    ui->toolButtonSubtitleFont->setBackgroundColor(QColor(s->value(QString("%1/SubtitleFontBackgroundColor").arg(classe),"white").toString()));
    ui->toolButtonTitleFont->setBackgroundColor(QColor(s->value(QString("%1/TitleFontBackgroundColor").arg(classe),"white").toString()));
    ui->toolButtonTocFont->setBackgroundColor(QColor(s->value(QString("%1/TocFontBackgroundColor").arg(classe),"white").toString()));
    ui->toolButtonCommentFont->setBackgroundColor(QColor(s->value(QString("%1/CommentFontBackgroundColor").arg(classe),"white").toString()));
    ui->toolButtonCommentItalicFont->setBackgroundColor(QColor(s->value(QString("%1/CommentItalicFontBackgroundColor").arg(classe),"white").toString()));
    ui->toolButtonCommentBoxFont->setBackgroundColor(QColor(s->value(QString("%1/CommentBoxFontBackgroundColor").arg(classe),"white").toString()));
    ui->toolButtonPageNumberFont->setBackgroundColor(QColor(s->value(QString("%1/PageNumberFontBackgroundColor").arg(classe),"white").toString()));

    ui->toolButtonCoverImage->setImage(s->value(QString("%1/CoverImage").arg(classe),"").toString());

    ui->colorButtonPaperColor->setColor(s->value(QString("%1/PaperColor").arg(classe),"#FFFFFF").toString());

    delete s;
}

void FormConfig::SetConfigFromFile(QString FileName)
{
    m_configFileName=FileName;
}

void FormConfig::SetConfigFromInit()
{
    m_configFileName="";
}


void FormConfig::Save(QString filename, Classes classe)
{
    QString section=classe2String(classe);
    QSettings sf(filename,QSettings::IniFormat);

    foreach (FontButton *w ,m_parent->findChildren<FontButton*>())
    {
        QRegExp tb("^toolButton");
        QRegExp f("Font$");
        QString name=w->objectName().replace(tb,"").replace(f,"");
        sf.setValue(QString("%1/%2Font").arg(section).arg(name),w->getFont().toString());
        sf.setValue(QString("%1/%2FontColor").arg(section).arg(name),w->getTextColor().name());
        sf.setValue(QString("%1/%2FontBackgroundColor").arg(section).arg(name),w->getBackgroundColor().name());
    }
    foreach (ColorButton *w ,m_parent->findChildren<ColorButton*>() )
    {
        QRegExp tb("^toolButton");
        QRegExp c("Color$");
        QString name=w->objectName().replace(tb,"").replace(c,"");
        sf.setValue(QString("%1/%2Color").arg(section).arg(name),w->getColor().name());
    }
    foreach (ImageButton *w ,m_parent->findChildren<ImageButton*>())
    {
        QRegExp tb("^toolButton");
        QRegExp i("Image$");
        QString name=w->objectName().replace(tb,"").replace(i,"");
        sf.setValue(QString("%1/%2Image").arg(section).arg(name),w->getImage());
    }
    foreach (QCheckBox *w ,m_parent->findChildren<QCheckBox*>())
    {
         QRegExp tb("^checkBox");
         QString name=w->objectName().replace(tb,"");
         sf.setValue(QString("%1/%2").arg(section).arg(name),w->isChecked()?"1":"0");
    }
    foreach (QComboBox *w ,m_parent->findChildren<QComboBox*>())
    {
         QRegExp tb("^comboBox");
         QString name=w->objectName().replace(tb,"");
         sf.setValue(QString("%1/%2").arg(section).arg(name),w->currentIndex());
    }
    foreach (QLineEdit *w ,m_parent->findChildren<QLineEdit*>())
    {
         QRegExp tb("^lineEdit");
         QString name=w->objectName().replace(tb,"");
         sf.setValue(QString("%1/%2").arg(section).arg(name),w->text());
    }
    foreach (SpinBoxUnit *w ,m_parent->findChildren<SpinBoxUnit*>())
    {
         QRegExp tb("^spu");
         QString name=w->objectName().replace(tb,"");
         sf.setValue(QString("%1/%2").arg(section).arg(name),w->valueunit());
    }
    foreach (QDoubleSpinBox *w ,m_parent->findChildren<QDoubleSpinBox*>())
    {
         QRegExp tb("^doubleSpinBox");
         QString name=w->objectName().replace(tb,"");
         sf.setValue(QString("%1/%2").arg(section).arg(name),w->value());
    }
    foreach (ColorButton *w ,m_parent->findChildren<ColorButton*>())
    {
         QRegExp tb("^colorButton");
         QString name=w->objectName().replace(tb,"");
         sf.setValue(QString("%1/%2").arg(section).arg(name),w->getColor().name());
    }
    sf.sync();
}



void FormConfig::displayFont ( QFont font, QColor text, QColor background)
{    QString name=sender()->objectName().replace("toolButton","");
    name=QString("label%1Example").arg(name);
    foreach (ExampleLabel *w ,m_parent->findChildren<ExampleLabel*>(name))
    w->setColor(font,text,background);

}


void FormConfig::displayThumb(QString image)
{

    QSettings *s;
    if ( m_configFileName.isEmpty() ) s= new QSettings;
    else s= new QSettings(m_configFileName);
    if (image.isEmpty()){ ui->labelCoverViewImage->setPixmap(QPixmap());return;}
    QPixmap pix(QString("%2").arg(image));
    QPixmap p=pix.scaledToWidth(150);
    ui->labelCoverViewImage->setPixmap(p);
    delete s;
}


void FormConfig::deleteCoverImage(bool)
{
    ui->labelCoverViewImage->setPixmap(QPixmap());
    ui->toolButtonCoverImage->setImage("");
}

QString FormConfig::classe2String(Classes name)
{
    if ( name == Memory ) return "Memory";
    if ( name == Text ) return "Text";
    if ( name == Lyrics) return "Lyrics";
    else  return "Chord";
}


void FormConfig::Retranslate()
{
    ui->retranslateUi(this);
}

void FormConfig::InitComboBoxChordInclusion(QStringList list)
{
    ui->comboBoxChordInText->clear();
    ui->comboBoxChordInText->addItems(list);
}
